patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -1543,15 +1551,43 @@
  			AllowedToSpreadInfections = true;
  			CreativePowers.StopBiomeSpreadPower power = CreativePowerManager.Instance.GetPower<CreativePowers.StopBiomeSpreadPower>();
  			if (power != null && power.GetIsUnlocked())
-@@ -45669,7 +_,7 @@
+@@ -45623,7 +_,7 @@
+ 				Liquid.skipCount = 0;
+ 			}
+ 
+-			int worldUpdateRate = GetWorldUpdateRate();
++			double worldUpdateRate = GetWorldUpdateRate();
+ 			if (worldUpdateRate == 0)
  				return;
  
- 			for (int l = 0; l < Main.dayRate; l++) {
+@@ -45668,8 +_,14 @@
+ 			if (Main.dayTime)
+ 				return;
+ 
+-			for (int l = 0; l < Main.dayRate; l++) {
 -				float num6 = Main.maxTilesX / 4200;
++			timePass += Main.desiredWorldEventsUpdateRate;
++			if (timePass < 1)
++				return;
++
++			double timePassReference = timePass;
++			for (int l = 0; l < timePassReference; l++) {
++				timePass -= 1.0;
 +				float num6 = Main.maxTilesX / 4200f; // Selfish fix for falling stars on extra small worlds
```

Comment:
This isn't your change, but I'm seeing that we are removing and than adding the same line. (1568 and 1576 on this).
Mind fixing?